docs/scos/dev/upgradability-services/upgradability-guidelines/private-api-is-used-on-the-project-level.md

```diff
@@ -0,0 +1,120 @@
+---
+title: Private API is used on the project level
+description: Reference information for evaluator and upgrader tools.
+last_updated: Mar 23, 2022
+template: concept-topic-template
+---
+
+Modules have public and private APIs. While public API updates always support backward compatibility, private API updates can break backward compatibility. So, backward compatibility is not guaranteed in the private API. For example, if you use a core method on the project level, and it is updated or removed, it can cause unexpected issues during updates.
+
+For more information about module APIs, see [Definition of Module API](/docs/scos/dev/architecture/module-api/definition-of-module-api.html).
+
+## Example of code that causes the upgradability error
+
+`CustomerFacade` uses `createCustomerReader` and `getCustomerCollection`. These are private API entities.
+
+```php
+namespace Pyz\Zed\Customer\Business;
+
+use Spryker\Zed\Customer\Business\CustomerFacade as SprykerCustomerFacade;
+
+/**
+ * @method \Pyz\Zed\Customer\Business\CustomerBusinessFactory getFactory()
+ */
+class CustomerFacade extends SprykerCustomerFacade implements CustomerFacadeInterface
+{
+   /**
+     * ...
+     */
+    public function getCustomerCollection(...): ...
+    {
+        return $this->getFactory()
+            ->createCustomerReader()
+            ->getCustomerCollection(...);
+    }
+}
+```
+
+## Example of related error in the Evaluator output
+
+```bash
+------------------------------------------------------------------------------------------------------------------------
+Pyz\Zed\Customer\Business\CustomerFacade
+"PrivateApi:Persistence Please avoid Spryker dependency: "{name}"
```

Comment:
Not sure that this error is correct in general. @VladislavStrelchenko could you please check it? 

I would assume that it will appear on some factory ot business level, not facade. 
Or will it appear on facade?

Reply:
Here I need some @VladislavStrelchenko or @alexanderM91 help. 
Guys, please take a look

Reply:
@helen-laktionova The message is an example of Evaluator output (Facade level)